sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpTransactionEnlistment.cs

```diff
@@ -31,13 +32,13 @@ internal class AmqpTransactionEnlistment : Singleton<AmqpTransactionEnlistment>,
 
         public ArraySegment<byte> AmqpTransactionId { get; private set; }
 
-        protected override async Task<AmqpTransactionEnlistment> OnCreateAsync(TimeSpan timeout)
+        protected override async Task<AmqpTransactionEnlistment> OnCreateAsync(TimeSpan timeout, CancellationToken cancellationToken)
```

Comment:
where does this cancellation token come from?

Reply:
The caller; this is part of `Singleton` which is the basis for `FaultTolerantAmqpObject`.   This would be called when someone requests the object via `GetOrCreateAsync` - the token passed to that would propagate here.

Reply:
But we still always use the GetOrCreate overload that takes a timeout, right?

Reply:
Does the AMQP lib create a CT from the timeout?

Reply:
> But we still always use the GetOrCreate overload that takes a timeout, right?

For creating our connections and links, yes.   The factory exists at a higher level than the method call that takes a token, so there's no straightforward way to find/pass the correct token.

> Does the AMQP lib create a CT from the timeout?

It does not; it passes `CancellationToken.None` if no token is specified.  _([src](https://github.com/Azure/azure-amqp/blob/hotfix/Microsoft.Azure.Amqp/Singleton.cs#L98))_     Timeout code is still using the `TimeoutHelper` approach.   _([src](https://github.com/Azure/azure-amqp/blob/hotfix/Microsoft.Azure.Amqp/Singleton.cs#L148))_